src/pages/shared/components/SelectGasMenu/SelectGasMenuBody.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { FC, useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { ConnectedProps, connect } from 'react-redux';
+import { useTranslation } from 'react-i18next';
+
+import { setGasPrice, getGasInfo } from 'ducks/transaction';
+import { RootState } from 'ducks/types';
+
+import NumericInput from 'components/Input/NumericInput';
+
+const mapStateToProps = (state: RootState) => ({
+	gasInfo: getGasInfo(state),
+});
+
+const mapDispatchToProps = {
+	setGasPrice,
+};
+
+const connector = connect(mapStateToProps, mapDispatchToProps);
+
+type PropsFromRedux = ConnectedProps<typeof connector>;
+
+type GasMenuProps = PropsFromRedux & {
+	setDropdownIsOpen: (isOpen: boolean) => void;
+};
+
+const SelectGasMenuBody: FC<GasMenuProps> = ({
+	setGasPrice,
+	gasInfo: { gasSpeed, gasPrice },
+	setDropdownIsOpen,
+}) => {
+	const { t } = useTranslation();
+	const [customGasPrice, setCustomGasPrice] = useState<number | undefined>(undefined);
+
+	const setGasPriceAndCloseDropdown = (updateGasPrice: number) => {
+		setGasPrice(updateGasPrice);
+		setDropdownIsOpen(false);
+	};
+
+	useEffect(() => {
+		if (customGasPrice !== undefined) {
+			setGasPrice(customGasPrice);
+		}
+	}, [setGasPrice, customGasPrice]);
+
+	return (
+		<Content>
+			<StyledNumericInput
+				id="1"
+				value={customGasPrice}
+				onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+					const newPrice = Number(e.target.value);
+					if (newPrice < 300) {
```

Comment:
I assume this is a safe value? if so, name it as a constant "MAX_SAFE_GAS" or something